staging/src/k8s.io/legacy-cloud-providers/gce/gce_disks.go

```diff
@@ -620,6 +624,49 @@ func (g *Cloud) DisksAreAttached(diskNames []string, nodeName types.NodeName) (m
 	return attached, nil
 }
 
+// BulkDisksAreAttached is a batch function to check if all corresponding disks are attached to the
+// nodes specified with nodeName.
+func (g *Cloud) BulkDisksAreAttached(diskByNodes map[types.NodeName][]string) (map[types.NodeName]map[string]bool, error) {
+	instanceNames := []string{}
+	for nodeName := range diskByNodes {
+		instanceNames = append(instanceNames, mapNodeNameToInstanceName(nodeName))
+	}
+
+	// List all instances with the given instance names
+	// Then for each instance listed, add the disks attached to that instance to a map
+	listedInstances, err := g.getInstancesByNames(instanceNames)
+	listedInstanceNamesToDisks := make(map[string][]*compute.AttachedDisk)
+	if err != nil && err != cloudprovider.InstanceNotFound {
```

Comment:
Handle error first (before allocating listedInstanceNamesToDisks)

Reply:
I moved it after the generic error handling but before the cloudprovider.InstanceNotFound handling because for InstanceNotFound to log the not found instances I need listedInstanceNamesToDisks.